opba-consent-rest-api/pom.xml

```diff
@@ -11,53 +11,105 @@
+      <groupId>javax.annotation</groupId>
+      <artifactId>javax.annotation-api</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.jetbrains</groupId>
-      <artifactId>annotations</artifactId>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
     </dependency>
-
-    <!-- Spring -->
     <dependency>
-      <groupId>org.springframework.boot</groupId>
-      <artifactId>spring-boot-starter-web</artifactId>
+      <groupId>javax.servlet</groupId>
+      <artifactId>javax.servlet-api</artifactId>
     </dependency>
-
-    <!-- test -->
     <dependency>
-      <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter</artifactId>
-      <version>${jupiter.version}</version>
-      <scope>test</scope>
+      <groupId>io.springfox</groupId>
+      <artifactId>springfox-swagger2</artifactId>
     </dependency>
+
     <dependency>
-      <groupId>org.junit.platform</groupId>
-      <artifactId>junit-platform-commons</artifactId>
-      <version>${junit.platform.commons.version}</version>
-      <scope>test</scope>
+      <groupId>io.springfox</groupId>
+      <artifactId>springfox-swagger-ui</artifactId>
     </dependency>
-
   </dependencies>
+  
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>io.swagger.codegen.v3</groupId>
+        <artifactId>swagger-codegen-maven-plugin</artifactId>
+        <version>3.0.11</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>generate</goal>
+            </goals>
+            <configuration>
+              <inputSpec>${openapi-spec-file-location}</inputSpec>
+              <language>spring</language>
+              <output>${project.basedir}/generated/</output>
+              <configOptions>
+                <sourceFolder>src/main</sourceFolder>
+                <basePackage>de.adorsys.opba.consent</basePackage>
+                <configPackage>de.adorsys.opba.consent.config</configPackage>
+                <modelPackage>de.adorsys.opba.consent.model</modelPackage>
+                <apiPackage>de.adorsys.opba.consent.resource</apiPackage>
+                <interfaceOnly>true</interfaceOnly>
+                <skipOverwrite>true</skipOverwrite>
+                <dateLibrary>java8</dateLibrary>
+                <java8>true</java8>
+                <implicitHeaders>false</implicitHeaders>
+                <delegatePattern>true</delegatePattern>
+                <useTags>true</useTags>
+                <library>spring-mvc</library>
+                <swaggerDocketConfig>false</swaggerDocketConfig>
+              </configOptions>
+            </configuration>
+          </execution>
+        </executions>
+        <dependencies>
+          <dependency>
+            <groupId>io.swagger.codegen.v3</groupId>
+            <artifactId>swagger-codegen-generators</artifactId>
+            <version>1.0.11</version>
+          </dependency>
+        </dependencies>
+      </plugin>
+    </plugins>
```

Comment:
Add 
```
<plugin>
				<groupId>org.codehaus.mojo</groupId>
				<artifactId>build-helper-maven-plugin</artifactId>
				<version>3.0.0</version>
				<executions>
					<execution>
						<id>add-generated-source</id>
						<phase>initialize</phase>
						<goals>
							<goal>add-source</goal>
						</goals>
						<configuration>
							<sources>
								<source>${project.basedir}/generated/src/main</source>
							</sources>
						</configuration>
					</execution>
				</executions>
			</plugin>
```
After io.swagger.codegen.v3 plugin, so that IDE can recognize it as generated code automatically.